src/ui.data-table/utilTable.js

```diff
@@ -15,13 +17,29 @@ export function getFilteredRow(row, columns) {
 export function syncRowCheck(row, selectedRows) {
   row.isChecked = selectedRows.includes(row.id);
 
-  if (row.row) {
+  if (row.row && !Array.isArray(row.row)) {
     row.row = syncRowCheck(row.row, selectedRows);
   }
 
   return row;
 }
 
+export function addRowId(row) {
+  const isRowId = typeof row.id !== "undefined" && row.id !== null && row.id !== "";
+
+  row.id = isRowId ? row.id : getRandomId();
+
+  if (row.row && !Array.isArray(row.row)) {
```

Comment:
```js
if (row.row) {
  row.row = Array.isArray(row.row) 
    ? row.row.map((nestedRow) => addRowId(nestedRow)) 
    : addRowId(row.row);
}
```

Reply:
That's just condition inside of another condition. Two separate if blocks looks muck cleaner and readable.

Reply:
`!Array.isArray(row.row)` and `Array.isArray(row.row)` in different conditions looks confusing to me. I prefer to keep conditon and ternary